### ISSUE TITLE
pylint doesn't know re.Pattern members

### ISSUE THREAD

**Issue by gerhard4** [PyCQA/pylint#7296]
### Bug description

This looks like perfectly valid Python to me, but not to pylint:

```python
"""pylint minimal example."""
import re
from dataclasses import dataclass

@dataclass
class _Test:
    pattern_instance: re.Pattern[str]

C = _Test(pattern_instance=re.compile(r'regex pattern'))
print(C.pattern_instance.pattern)  # E1101: Instance of 'Pattern' has no 'pattern' member (no-member)
```

If I don't go through a dataclass with the variable, it seems pylint is smart enough to know what's going on. But not when it's set up like this.

### Command used

```shell
pylint pylint_pattern.py
```

### Pylint output

```shell
************* Module pylint_pattern
pylint_pattern.py:10:6: E1101: Instance of 'Pattern' has no 'pattern' member (no-member)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.5 (main, Jun 15 2022, 13:20:57) [GCC 9.4.0]
```

### OS / Environment

Windows 11, WSL (Ubuntu)

**Comment by clavedeluna** [comment]
This isn't reproducible on every python version. I could repro it on 3.11.0 and reporter used 3.10.5.

Using Python 3.8.9 I get a different result:

```
test.py:6:22: E1136: Value 're.Pattern' is unsubscriptable (unsubscriptable-object)
```

**Comment by acabal33uab** [comment]
Still occurs with python 3.14
